tests/test_execute.py

```diff
@@ -0,0 +1,252 @@
+from myst_nb.cache import execution_cache, add_notebook_outputs
+from myst_nb.parser import NotebookParser
+from myst_nb.nb_glue.domain import NbGlueDomain
+
+from docutils import nodes
+from docutils.frontend import OptionParser
+from docutils.parsers.rst import Parser as RSTParser
+from docutils.utils import new_document
+
+from sphinx.domains.math import MathDomain
+from sphinx.util.docutils import sphinx_domains
+
+from nbdime.diffing.notebooks import (
+    diff_notebooks,
+    set_notebook_diff_targets,
+    set_notebook_diff_ignores,
+)
+from nbdime.prettyprint import pretty_print_diff
+
+import nbformat as nbf
+from collections import defaultdict
+
+import pytest
+import os
+
+# -Diff Configuration-#
+NB_VERSION = 4
+set_notebook_diff_ignores({"/nbformat_minor": True})
+set_notebook_diff_targets(metadata=False)
+
+
+class MockEnv:
+    def __init__(self, tmp_path):
+        self.docname = "source/nb"
+        self.dependencies = defaultdict(set)
+        self.domaindata = {}
+        self.domains = {
+            NbGlueDomain.name: NbGlueDomain(self),
+            MathDomain.name: MathDomain(self),
+        }
+        self._tmp_path = tmp_path
+        self.path_cache = str(tmp_path) + "/.jupyter_cache"
+
+        class app:
+            class builder:
+                name = "html"
+
+            outdir = str(tmp_path) + "/build" + "/outdir"
+
+        class env:
+            srcdir = str(tmp_path) + "/source"
+            outdir = str(tmp_path) + "/build" + "/outdir"
+            config = {
+                "jupyter_execute_notebooks": "cache",
+                "execution_excludepatterns": [],
+                "jupyter_cache": "",
+            }
+
+        self.app = app
+        self.env = env
+        self.env.app = app
+        self.config = env.config
+
+    def set_config(self, config_val):
+        for key, val in self.env.config.items():
+            if config_val[0] == key:
+                self.env.config[key] = config_val[1]
+
+    def set_srcdir(self, srcdir):
+        self.env.srcdir = srcdir
+
+    def set_docname(self, docname):
+        self.docname = docname
+
+
+@pytest.fixture()
+def mock_environment(tmp_path):
+    env = MockEnv(tmp_path)
+    return env
+
+
+@pytest.fixture()
+def mock_document(tmp_path) -> nodes.document:
+    settings = OptionParser(components=(RSTParser,)).get_default_values()
+    document = new_document("notset", settings=settings)
+    document.settings.env = MockEnv(tmp_path)
+    with sphinx_domains(document.settings.env):
+        yield document
+
```

Comment:
This can all be moved to `conftest.py` and merged with the largely duplicated code there (if you didn't know, you can use test fixtures defined in conftest.py in any test file with the same root path)

Reply:
@AakashGfude do you wanna try making this simplification to the tests, and then if they're happy we can merge?

Reply:
@choldgraf @chrisjsewell  sure had separated it to play around while not affecting other tests. Thanks for the reminder.